### ISSUE TITLE
flattenEvent() fails with "'ascii' codec can't encode character" in Python 2.7

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8699]
|[<img alt="znerol's avatar" src="https://avatars.githubusercontent.com/u/23288?s=50" width="50" height="50">](https://github.com/znerol)| @znerol reported|
|-|-|
|Trac ID|trac#8699|
|Type|defect|
|Created|2016-07-28 05:59:35Z|

backtrace ends in _flatten.py

```
twisted/logger/_flatten.py(119)flattenEvent()
-> flattenedValue = conversionFunction(fieldValue)
(Pdb) conversionFunction
<type 'str'>
```

I guess conversionFunction should be unicode in Python 2.7 and str in Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__8699 8699
type__defect defect
reporter__znerol znerol
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469685575533160 1469685575533160
changetime__1470085969851774 1470085969851774
version__None None
owner__Craig_Rodrigues__rodrigc_____ Craig Rodrigues <rodrigc@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="znerol's avatar" src="https://avatars.githubusercontent.com/u/23288?s=50" width="50" height="50">](https://github.com/znerol)<a name="note_1"></a>|@znerol commented|
|-|-|

PR: https://github.com/twisted/twisted/pull/437

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc set owner to @rodrigc<br>@rodrigc set status to `closed`|
|-|-|

In changeset 44579f4fbd6c71c65ad0bf11a576b9124c8a13f3
```
#!CommitTicketReference repository="" revision="44579f4fbd6c71c65ad0bf11a576b9124c8a13f3"
Fix #8699 ascii encoding error in flattenEvent()
```

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub commented|
|-|-|

In changeset b636ffb4afba9f1c30b91ddef95e4cef38c2dd26
```
#!CommitTicketReference repository="" revision="b636ffb4afba9f1c30b91ddef95e4cef38c2dd26"
Merge pull request #452 from twisted/8699-znerol-ascii-encoding-error-in-flatten-event

Author: znerol
Reviewer: wsanchez, rodrigc
Fixes: #8699

Fix ascii encoding error in flattenEvent()
```
